### ISSUE TITLE
add fields to form-fields

### ISSUE THREAD

**Issue by glettl** [payloadcms/plugin-form-builder#35]
Currently it is possible to overwrite the fields of the form-field blocks completely, but it is not possible to add an additional custom field.

If I'd like to add the custom field 'test', all other fields are gone.

`formBuilder({
      fields: {
        email: {
          fields: [
            {
              name: 'test',
              type: 'text',
            },
          ],
        },
      },
    })`

I think, it's because the the merge in https://github.com/payloadcms/plugin-form-builder/blob/199771947d512ac227ddbcff12c190743092b534/src/collections/Forms/index.ts#L97
is configured to replace the fields array.

I know that there are cases where one wants to redefine all form-fields, however it would be nice if a way to simply add custom fields to an form-field could be provided.

I thought about two possible ways to achieve this:

1.  add an additional property '_addFields_' to the formBuilder incomingFormConfig.field.[FIELD_SLUG] object and add these fields after the merge step of the _generateFormCollection_ function.
2. expose the definition of the default form-fields configuration so that it can be imported and be reused the custom config. E.g. like this:
`import formFields from '@payloadcms/plugin-form-builder/';
[...]
formBuilder({
      fields: {
        email: {
          fields: [
            ...formFields.email.fields,
            {
              name: 'test',
              type: 'text',
            },
          ],
        },
      },
    })
`

**Comment by jacobsfletch** [comment]
Hey @glettl what you're describing makes sense to me. And you're right, we should allow for additions to made to _the fields within each block_ not just the block itself. But I think the API could be simpler than what you propose, we're already merging the partial block config in with the plugin's default block, so it sounds like we just need to take this one step further and merge the _fields_ within the block, too.

```ts
fields: {
  email: {
    fields: [...] // merge this instead of replace
  }
}
```

What do you think of this?

**Comment by jacobsfletch** [comment]
> I know that there are cases where one wants to redefine all form-fields, however it would be nice if a way to simply add custom fields to an form-field could be provided.

Yep ok I see exactly why you proposed your two solutions. Either add a flag to set the merge strategy, or import the fields to merge them yourself. It does feel like _merge_ should have been the default, but definitely don't want to introduce breaking changes. So let's just export the fields like you propose (option 2) this way we don't have to change the API at all.

**Comment by jacobsfletch** [comment]
@glettl I just opened PR #39 for this, let me know what you think!

**Comment by jacobsfletch** [comment]
This was just published in `1.0.14-canary.0` 🎉

**Comment by jacobsfletch** [comment]
And now `v1.0.14` 🚀
